packages/image_picker/image_picker_for_web/lib/image_picker_for_web.dart

```diff
@@ -310,6 +327,73 @@ class ImagePickerPlugin extends ImagePickerPlatform {
     _target.children.add(element);
     element.click();
   }
+
+  Future<String> _getCompressedUri(html.File file, int? imageQuality,
+      double? maxHeight, double? maxWidth) async {
+    //max width, max height, imageQuality are not supported for gif
+    if (file.type == "image/gif") {
+      return html.Url.createObjectUrl(file);
+    }
+    final Completer<String> _completer = Completer<String>();
+    final blobUrl = html.Url.createObjectUrl(file);
+    final image = html.ImageElement(src: blobUrl);
+    image.onLoad.listen((event) async {
+      html.Url.revokeObjectUrl(blobUrl);
+      final canvas = html.CanvasElement();
+      final size = _calculateSize(image, maxWidth ?? image.width!.toDouble(),
+          maxHeight ?? image.height!.toDouble());
+      canvas.width = size[0];
+      canvas.height = size[1];
+      final ctx = canvas.context2D;
+      if (maxHeight == null && maxWidth == null) {
+        ctx.drawImage(image, 0, 0);
+      } else {
+        ctx.drawImageScaled(image, 0, 0, canvas.width!, canvas.height!);
+      }
+      final blob = await canvas.toBlob(
+          file.type,
+          (imageQuality ?? 100) /
+              100.0); // Image quality only works for jpeg images
+      _completer.complete(html.Url.createObjectUrlFromBlob(blob));
+    });
+    image.onError.listen((event) {
+      //Return the original image if error comes
+      _completer.complete(html.Url.createObjectUrl(file));
+    });
+    return _completer.future;
+  }
+
+  List<int> _calculateSize(
+      html.ImageElement img, double maxWidth, double maxHeight) {
```

Comment:
Move this helper to the `ImageResizer` object as well. Also, maybe return a `Size` object [docs](https://api.flutter.dev/flutter/dart-ui/Size-class.html), instead of a list of ints?